homeassistant/components/deconz/alarm_control_panel.py

```diff
@@ -0,0 +1,133 @@
+"""Support for deCONZ alarm control panel devices."""
+from __future__ import annotations
+
+from pydeconz.sensor import (
+    ANCILLARY_CONTROL_ARMED_AWAY,
+    ANCILLARY_CONTROL_ARMED_NIGHT,
+    ANCILLARY_CONTROL_ARMED_STAY,
+    ANCILLARY_CONTROL_DISARMED,
+    AncillaryControl,
+)
+
+from homeassistant.components.alarm_control_panel import (
+    DOMAIN,
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+    SUPPORT_ALARM_ARM_NIGHT,
+    AlarmControlPanelEntity,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import NEW_SENSOR
+from .deconz_device import DeconzDevice
+from .gateway import get_gateway_from_config_entry
+
+DECONZ_TO_ALARM_STATE = {
+    ANCILLARY_CONTROL_ARMED_AWAY: STATE_ALARM_ARMED_AWAY,
+    ANCILLARY_CONTROL_ARMED_NIGHT: STATE_ALARM_ARMED_NIGHT,
+    ANCILLARY_CONTROL_ARMED_STAY: STATE_ALARM_ARMED_HOME,
+    ANCILLARY_CONTROL_DISARMED: STATE_ALARM_DISARMED,
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities) -> None:
+    """Set up the deCONZ alarm control panel devices.
+
+    Alarm control panels are based on the same device class as sensors in deCONZ.
+    """
+    gateway = get_gateway_from_config_entry(hass, config_entry)
+    gateway.entities[DOMAIN] = set()
+
+    @callback
+    def async_add_alarm_control_panel(sensors=gateway.api.sensors.values()) -> None:
+        """Add alarm control panel devices from deCONZ."""
+        entities = []
+
+        for sensor in sensors:
+
+            if (
+                sensor.type in AncillaryControl.ZHATYPE
```

Comment:
What does this do?

Reply:
deCONZ has two main resource types sensors and lights. These have later evolved into supporting many different types of devices. In this particular case it filters out sensors that are of the type ancillary control. AncillaryControl is a subclass of deconz sensors and ZHATYPE is a tuple with the different types that fit into this device type.

https://github.com/Kane610/deconz/blob/ac8b386ca6913c487a3277f511b90b4fc1064204/pydeconz/sensor.py#L179